input location support

milovate opened the pull request:
### Legal Boilerplate

Look, I get it. The entity doing business as “Gooey.AI” and/or “Dara.network” was incorporated in the State of Delaware in 2020 as Dara Network Inc. and is gonna need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Dara Network Inc can use, modify, copy, and redistribute my contributions, under its choice of terms.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added the ability for users to share their geographic location via a modal with map selection, supporting both Google Maps and OpenStreetMap.
  - Outgoing messages can now display embedded maps when location data is present.
  - Introduced new location-related icons and a spinner icon for improved visual feedback.
- **Bug Fixes**
  - Disabled outlined buttons no longer display hover effects for a more consistent user experience.
- **Style**
  - Added and updated styles for the location modal and buttons to enhance usability and appearance.
- **Chores**
  - Updated dependencies to support new map features.
- **Documentation**
  - Minor comment corrections for clarity.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

---
## EntelligenceAI PR Summary 
 Adds location sharing to the chat widget with map selection and preview features.
- Added new SVG icons for location and loading (src/assets/SvgIcons/)
- Introduced LocationModal for map-based location selection (src/widgets/copilot/components/Messages/LocationModal.tsx, .scss)
- Updated IncomingMsg to handle location button and modal (src/widgets/copilot/components/Messages/IncomingMsg.tsx)
- OutgoingMsg now displays a map preview for sent locations (src/widgets/copilot/components/Messages/OutgoingMsg.tsx)
- Extended config types for Google Maps API key (src/contexts/types.ts)
- Updated dependencies for mapping support (package.json, package-lock.json)
- Minor comment fix in Sources.tsx

milovate committed:
geolocation support

milovate commented on src/widgets/copilot/components/Messages/OutgoingMsg.tsx:
```diff
@@ -4,14 +4,43 @@ import { memo } from "react";
+  const zoomLevel = 16;
+  let bbox = '';
+  if (latitude !== undefined && longitude !== undefined) {
+    input_prompt="";
```
If a location request is made, only the iframe is displayed, not the button title.

milovate commented on src/widgets/copilot/components/Messages/IncomingMsg.tsx:
```diff
@@ -89,13 +89,65 @@ const FeedbackButtons = ({
-                          });
+                           // Check if this is a location request button
+                           if (button.id.includes('send_location')) {
+                            if (navigator.geolocation) {
```
referred to `button.id.includes("thumb")`

milovate committed:
add: Location modal popup

milovate committed:
feat: Add location-picker with Google Maps
- Adds overlay and map support for openstreetmaps
- Fixes zoom issue in LocationModal

devxpy commented:
@anish-work can you do the first round of review here

devxpy commented:
@coderabbitai full review

devxpy commented:
@entelligenceai

devxpy commented:
@review_pr

milovate commented on src/widgets/copilot/components/Messages/LocationModal.tsx:
```diff
@@ -0,0 +1,443 @@
+        <div className={clsx("location-map error-text")}>
+          {error ? (
+            <p className="p-4 text-center text-red-600">{error}</p>
+          ) : hasGoogleMapsKey ? (
+            renderGoogleMap()
+          ) : (
+            renderLeafletMap()
+          )}
```
opted not to automatically fall back to Leaflet. assuming we want users to be immediately aware of and debug any issues with their Google Maps API key.

milovate commented on src/assets/SvgIcons/LocationMarker.tsx:
```diff
@@ -0,0 +1,34 @@
+import SvgIcon from "src/components/shared/SvgIcon";
+
+const LocationMarker = (props: any) => {
```
consistent with other svg icons

milovate commented on src/assets/SvgIcons/LocationRefresh.tsx:
```diff
@@ -0,0 +1,22 @@
+import SvgIcon from "src/components/shared/SvgIcon";
+
+const LocationRefresh = (props: any) => {
```
consistent with other svg icons

milovate committed:
fix: Update iframe positioning and add lazy loading for location map

anish-work commented:
@milovate are you still working on this? I can see a black shadow at bottom when having conversation with the widget I suspect it's the Modal's shadow?

milovate commented:
> @milovate are you still working on this? I can see a black shadow at bottom when having conversation with the widget I suspect it's the Modal's shadow?
i've cleaned up the black shadow issue and amended the commit. Let me know if there's anything else I can fix up, aside from the secondary drawer

devxpy committed:
fix: wrap center coords always between -180 and +180 degrees.

refactor:
- Extracted OpenStreetMap embed URL logic into buildOpenStreetMapEmbedUrl function
- Refactored LocationModal to use imperative API and improved map handling (Google Maps/Leaflet)
- Updated context types to use secrets with GOOGLE_MAPS_API_KEY
- Improved feedback button logic and UI in IncomingMsg
- Moved and refactored calculateBoundingBox to OutgoingMsg
- Separtated inline widgets in location modal with top line functions

anish-work committed:
render normalButtons even if no thumb butons

anish-work committed:
tweak location styles

anish-work committed:
remane location arrow icon

anish-work committed:
dry out refresh location code

anish-work committed:
add fallback for location button with no content

anish-work committed:
add tooltip to refresh location button

devxpy committed:
feat: Enhance GoogleMapView with location search functionality
- Added input field for location search using Google Places Autocomplete
- Implemented event listener to update map and selected location based on user input
- Updated APIProvider to load Google Maps Places library

devxpy commented:
<img src="https://github.com/user-attachments/assets/3969c3c6-0535-48d7-ae9f-a867e439f323" alt="LGTM" height=400 width=auto />

devxpy committed:
fix: scroll wheel zoom should be centered

devxpy committed:
fix: autocomplete not showing up on fullscreen
refactor: encapsulate googleMapsError into GoogleMapView component

milovate commented on src/widgets/copilot/components/Messages/index.tsx:
```diff
@@ -18,25 +18,20 @@ const Responses = (props: any) => {
-        const role = responseData.role;
-        if (role === "user")
+        if (responseData.role === "user") {
+          return <OutgoingMsg key={id} />;
```
OutgoingMsg does not contain `responseData` props

devxpy replied:
@anish-work this bug was not caught by linter because the `Responses` widget is not typed correctly

milovate committed:
fix: pass complete message data to OutgoingMsg component
